WordPress/gutenberg
# Social Icon: Try auto-matching variation based on URL

## What?
Resolves #56259.

PR updates Social Icon's block to auto-match service variation based on the URL pattern. It also enables direct insertion of the block.

> [!NOTE]  
> Some variations like Mastodon, which allow services to be hosted on different domains, can't be auto-matched. But they can be inserted using the global block inserter.

Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com> 

## Why?
See #56259.

## How?
 Define pattern matches for each variation, similar to how Embeds block implementation.

## Testing Instructions
1. Open a post or page.
2. Insert the Social Icons block.
3. Add a social icon.
4. Type URLs for a few social networks.
5. The block's variation should change with URLs.
6. When a URL doesn't match the variation, it should display a link icon.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/240569/2e91dd07-7e73-45d1-aad5-39ac02e452d5
